src/protocols/rdp/settings.c

```diff
@@ -1189,6 +1189,9 @@ void guac_rdp_push_settings(guac_client* client,
     rdp_settings->DesktopHeight = guac_settings->height;
     rdp_settings->AlternateShell = guac_rdp_strdup(guac_settings->initial_program);
     rdp_settings->KeyboardLayout = guac_settings->server_layout->freerdp_keyboard_layout;
+    rdp_settings->KeyboardType = guac_settings->server_layout->freerdp_keyboard_type;
+    rdp_settings->KeyboardSubType = guac_settings->server_layout->freerdp_keyboard_subtype;
+    rdp_settings->KeyboardFunctionKey = guac_settings->server_layout->freerdp_keyboard_function_key;
```

Comment:
Will this cause problems for the keymap files that don't have these values defined?

Reply:
Sorry, it should not be set if no value.